### PR TITLE
Readme markdown style compliance

### PR DESCRIPTION
**Pull request by kotp**
No squash.

### CODE REVIEW

**Comment by kotp** [comment]
@exercism/reviewers

File: README.md [hunk 1]
```diff
@@ -384,22 +322,49 @@ Use the `configlet` tool to generate a README from shared metadata:
...
+
 The Ruby icon is the Vienna.rb logo, and is used with permission. Thanks Floor
 Dress :)
 
+[1]: https://github.com/exercism/ruby/workflows/Configlet%20CI/badge.svg
+[2]: https://github.com/exercism/ruby/actions?query=workflow%3A%22Configlet+CI%22+branch%3Amaster
+[3]: https://github.com/exercism/ruby/workflows/Exercise%20tests/badge.svg
+[4]: https://github.com/exercism/ruby/actions?query=workflow%3A%22Exercise+tests%22+branch%3Amaster
+[5]: https://badges.gitter.im/exercism/ruby.svg
+[6]: https://gitter.im/exercism/ruby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge
```
Comment (ErikSchierboom) [inline README.md:342]: Why use numbers here instead of text?

Comment (kotp) [reply README.md:342]: I'mage links I think was my motivation.  (The badge.svg repeats, so this avoids that.)

Comment (kotp) [reply README.md:342]: Yeah, it was the "badge area" and so not really all that relevant to the content, like the other links.

Comment (ErikSchierboom) [reply README.md:342]: I'd still prefer text I think:

```suggestion
[configlet-workflow-badge]: https://github.com/exercism/ruby/workflows/Configlet%20CI/badge.svg
[configlet-workflow]: https://github.com/exercism/ruby/actions?query=workflow%3A%22Configlet+CI%22+branch%3Amaster
[tests-workflow-badge]: https://github.com/exercism/ruby/workflows/Exercise%20tests/badge.svg
[tests-workflow]: https://github.com/exercism/ruby/actions?query=workflow%3A%22Exercise+tests%22+branch%3Amaster
[ruby-gitter-badge]: https://badges.gitter.im/exercism/ruby.svg
[ruby-gitter-channel]: https://gitter.im/exercism/ruby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge
```

**Review by ErikSchierboom — DISMISSED** [review]
Great work!

**Review by kotp — COMMENTED** [review]
Rendering ["Proof"][1] in case there is no mechanism locally for reviewing.

[1]: https://github.com/exercism/ruby/tree/README_markdown_style_compliance#readme

File: README.md [hunk 2]
```diff
@@ -384,22 +322,50 @@ Use the `configlet` tool to generate a README from shared metadata:
...
+[#ruby-icon]: #ruby-icon
+[#running-the-tests]: #running-the-tests
+[rubocop]: http://batsov.com/rubocop/
+[ruby-gitter-badge]: https://badges.gitter.im/exercism/ruby.svg
+[ruby-gitter-channel]: https://gitter.im/exercism/ruby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge
+[#setup]: #setup
+[#style-guide]: #style-guide
+[synchronize]: https://help.github.com/articles/syncing-a-fork/
+[tests-workflow-badage]: https://github.com/exercism/ruby/workflows/Exercise%20tests/badge.svg
+[tests-workflow]: https://github.com/exercism/ruby/actions?query=workflow%3A%22Exercise+tests%22+branch%3Amaster
```
Comment (ErikSchierboom) [inline README.md]: There was a typo in the first link. I've also changed to use the badge links that GitHub suggests (the second like was also using the deprecated `master` branch)

```suggestion
[tests-workflow-badge]: https://github.com/exercism/ruby/actions/workflows/exercise-tests.yml/badge.svg
[tests-workflow]: https://github.com/exercism/ruby/actions/workflows/exercise-tests.yml
```

File: README.md [hunk 3]
```diff
@@ -384,22 +322,50 @@ Use the `configlet` tool to generate a README from shared metadata:
...
 Dress :)
 
+[#anatomy-of-an-exercise]: #anatomy-of-an-exercise
+[assertion]: https://github.com/exercism/ruby/blob/master/lib/generator/exercise_case/assertion.rb
+[#canonical-data]: #canonical-data
+[canonical data]: https://github.com/exercism/problem-specifications/tree/master/exercises
+[canonical exercises]: https://github.com/exercism/problem-specifications/tree/master/exercises
+[#changing-a-generated-exercise]: #changing-a-generated-exercise
+[configlet-workflow-badge]: https://github.com/exercism/ruby/workflows/Configlet%20CI/badge.svg
+[configlet-workflow]: https://github.com/exercism/ruby/actions?query=workflow%3A%22Configlet+CI%22+branch%3Amaster
```
Comment (ErikSchierboom) [inline README.md:344]: I've changed the links to use the badge links that GitHub suggests (the second like was also using the deprecated master branch)

```suggestion
[configlet-workflow-badge]: https://github.com/exercism/ruby/actions/workflows/configlet.yml/badge.svg
[configlet-workflow]: https://github.com/exercism/ruby/actions/workflows/configlet.yml
```

**Review by ErikSchierboom — APPROVED** [review]

**Comment by kotp** [comment]
Thanks @ErikSchierboom

**Pull request merged**